src/nvim/eval.c

```diff
@@ -14872,6 +14878,12 @@ static void f_setreg(typval_T *argvars, typval_T *rettv, FunPtr fptr)
                           append, yank_type, block_len);
   }
   rettv->vval.v_number = 0;
+
+  if (set_unnamed) {
+    if (!op_register_set_previous(regname)) {
+      EMSG("Unable to set previous yank register");
```

Comment:
Thinking about this more, `f_setreg` already prints an error in the case where `op_register_set_previous` would fail (when `regname` isn't a valid name it errors with `E354`). So it would make sense to discard `op_register_set_previous`'s result and not print a separate error message.